### ISSUE TITLE
[Feature]: add support to connect with usb-serial instead of pseudo-tty

### ISSUE THREAD

**Issue by rohoog** [termux/termux-app#3429]
### Feature description

termux is a nice terminal emulator, but it can only 'talk' to a linux shell program. Android doesn't normally (unless rooted) allow to use the /dev/tty{USB,ACM}x device nodes. There is a library https://github.com/mik3y/usb-serial-for-android that adds this using the Android permissions.

It would be a great added value to allow this, so we can use termux to talk to our USB gadgets built with microcontroller boards, such as Arduino/STM32 etc. using USB-OTG, so we can do software maintenance using the smartphone without the effort of writing an app.

### Additional information

SerialUSBTerminal (Kai Morich) does this, but it doesn't have proper terminal emulation; it looks like the arduino-IDE 'serial monitor'.

**Comment by idarlund** [comment]
Found this thread and I'd also very much would like this feature!
Not sure if this might be out of scope by termux, but either way it would be a great feature to add!

**Comment by MarkWllms** [comment]
I also agree. AnotherTerm does this natively. I have compiled the slirp PPP emulator to provide 'dial-up' over serial in Termux. At the moment, I have to use the TCPUART app and socat to get the serial line to the emulated PPP server.

I have managed to get a version of this working natively using Termux-usb, libusb and the USBUART C++ library, but it would be much easier if it could all be done on the Android side by Termux, or maybe by Termux-API.

**Comment by HusniMuhammad** [comment]
Compiling tools available in termux


Use  esp module  runs esp-link to flash Arduino avr arm microcontrollers
